services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -223,8 +223,6 @@ public void run()
             )
         );
         injector.getInstance(ExecutorLifecycle.class).join();
-        // Explicitly call lifecycle stop, dont rely on shutdown hook.
-        lifecycle.stop();
```

Comment:
Quite simply: Because I'm trying to push the nodes to be able to have a very clean shutdown. That means all threads exit properly on their own accord or are daemon threads, and that shutdown hooks are properly executed.

For a thread to hold up the entire JVM (excluding the main thread) is probably a bug.

We don't call lifecycle.stop() on any of the other CLI's and I don't think it is reasonable to require any module or custom CLI to explicitly call lifecycle.stop(), but only if it intends to eventually exit.

This is mostly to help shake out any potential bugs where we may not have fixed a daemon thread.

You are correct that it is harder to test. And if you think this PR should be blocked until the IT are fixed, then we can talk about what the expectations are (I want to fix the IT for this kind of stuff anyways).